package.json

```diff
@@ -8,99 +8,91 @@
   "bugs": {
     "url": "https://github.com/mattermost/mattermost-redux/issues"
   },
+  "main": "build/mattermost.client4.js",
+  "types": "build/index.d.ts",
+  "files": [
+    "build"
```

Comment:
When I include this branch in mattermost-webapp some of the expected files are missing at the tests fail. I think it has to do with this line. Do you have a branch of mattermost-webapp that works with this mattermost-redux branch.

Reply:
@anindha - can you elaborate on "some of the expected files are missing at the tests fail" ?
If you have some error logs - please paste them here.
This branch should work with regular webapp without changes

Reply:
@reflog Here is an example. Could be my setup. Currently pointing webapp master at your branch. This is my package.json:
`"mattermost-redux": "github:reflog/mattermost-redux#typescript",`


`FAIL  components/channel_header_mobile/channel_info_button/channel_info_button.test.js
  ● Test suite failed to run

    Cannot find module 'react-hot-loader' from 'permissions.js'

    However, Jest was able to find:
        './channel_info_button.js'
        './channel_info_button.test.js'

    You might want to include a file extension in your import, or update your 'moduleFileExtensions', which is currently ['ts', 'tsx', 'js', 'jsx', 'json'].

    See https://jestjs.io/docs/en/configuration#modulefileextensions-array-string

      2 | 
      3 | Object.defineProperty(exports, "__esModule", {
    > 4 |   value: true
        |                                  ^
      5 | });
      6 | exports.default = void 0;
      7 | // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:230:17)
      at ../mattermost-redux/constants/permissions.js:4:91
      at Object.<anonymous> (../mattermost-redux/constants/permissions.js:6:3)`

Reply:
Here is an error in the console when running the `npm run dev-server`. The app fails to run.

```client?9298:159 ./actions/websocket_actions.jsx
Module not found: Error: Can't resolve 'mattermost-redux/action_types' in '/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/actions'```

Reply:
@anindha - thanks for the logs! I've fixed the issue, can you try now?

Reply:
@reflog Still having the issue

Reply:
Can you provide more details please?

Reply:
Seems to be a problem how the include path is being resolved. Here is an example.

```
ERROR in ./components/user_settings/security/user_access_token_section/user_access_token_section.jsx
Module not found: Error: Can't resolve 'mattermost-redux/utils/user_utils' in '/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/security/user_access_token_section'
resolve 'mattermost-redux/utils/user_utils' in '/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/security/user_access_token_section'
  Parsed request is a module
  using description file: /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/package.json (relative path: ./components/user_settings/security/user_access_token_section)
    resolve as module
      /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/security/user_access_token_section/node_modules doesn't exist or is not a directory
      /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/security/node_modules doesn't exist or is not a directory
      /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/node_modules doesn't exist or is not a directory
      /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/node_modules doesn't exist or is not a directory
      /Users/anindha/go/src/github.com/mattermost/node_modules doesn't exist or is not a directory
      /Users/anindha/go/src/github.com/node_modules doesn't exist or is not a directory
      /Users/anindha/go/src/node_modules doesn't exist or is not a directory
      /Users/anindha/go/node_modules doesn't exist or is not a directory
      /Users/anindha/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules
        using description file: /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/package.json (relative path: ./node_modules)
          using description file: /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/package.json (relative path: ./utils/user_utils)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.tsx doesn't exist
            as directory
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils doesn't exist
      looking for modules in /Users/anindha/go/src/github.com/mattermost/mattermost-webapp
        using description file: /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/package.json (relative path: .)
          using description file: /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/package.json (relative path: ./mattermost-redux/utils/user_utils)
            no extension
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils doesn't exist
            .js
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.js doesn't exist
            .jsx
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.jsx doesn't exist
            .ts
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.ts doesn't exist
            .tsx
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.tsx doesn't exist
            as directory
              /Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils doesn't exist
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/security/user_access_token_section/node_modules]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/security/node_modules]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/user_settings/node_modules]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/components/node_modules]
[/Users/anindha/go/src/github.com/mattermost/node_modules]
[/Users/anindha/go/src/github.com/node_modules]
[/Users/anindha/go/src/node_modules]
[/Users/anindha/go/node_modules]
[/Users/anindha/node_modules]
[/Users/node_modules]
[/node_modules]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.js]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.jsx]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.ts]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/node_modules/mattermost-redux/utils/user_utils.tsx]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.js]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.jsx]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.ts]
[/Users/anindha/go/src/github.com/mattermost/mattermost-webapp/mattermost-redux/utils/user_utils.tsx]
 @ ./components/user_settings/security/user_access_token_section/user_access_token_section.jsx 11:0-63 127:11-34
 @ ./components/user_settings/security/user_access_token_section/index.js
 @ ./components/user_settings/security/user_settings_security.jsx
 @ ./components/user_settings/security/index.js
 @ ./components/user_settings/user_settings.jsx
 @ ./node_modules/babel-loader/lib??ref--4!./components/user_settings/index.js
 @ ./node_modules/bundle-loader?lazy!./components/user_settings/index.js
 @ ./components/user_settings/modal/user_settings_modal.jsx
 @ ./components/user_settings/modal/index.js
 @ ./components/sidebar/header/dropdown/sidebar_header_dropdown.jsx
 @ ./components/sidebar/header/dropdown/index.js
 @ ./components/sidebar/header/sidebar_header.jsx
 @ ./components/sidebar/header/index.js
 @ ./components/sidebar/sidebar.jsx
 @ ./components/sidebar/index.js
 @ ./components/channel_layout/channel_controller.jsx
 @ ./components/needs_team/needs_team.jsx
 @ ./components/needs_team/index.js
 @ ./components/root/root.jsx
 @ ./node_modules/babel-loader/lib??ref--4!./components/root/index.js
 @ ./node_modules/bundle-loader?lazy!./components/root/index.js
 @ ./components/app.jsx
 @ ./root.jsx
 @ multi ./root.jsx root.html
```